File: backend/src/main/java/com/mini_erp/backend/supplier/domain/Supplier.java
package com.mini_erp.backend.supplier.domain;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import java.time.LocalDateTime;

@Entity
@Table(name = "suppliers")
@Getter @Setter
public class Supplier {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 200)
    private String name;

    @Column(nullable = false, length = 15)
    private String nip;

    @Column(nullable = false, length = 150)
    private String email;

    @Column(nullable = false, length = 30)
    private String phone;

    @Column(nullable = false, length = 200)
    private String street;

    @Column(nullable = false, length = 100)
    private String city;

    @Column(name = "postal_code", nullable = false, length = 10)
    private String postalCode;

    @Column(nullable = false, length = 60)
    private String country;

    @Column(nullable = false)
    private boolean active = true;

    @Column(name = "created_at", insertable = false, updatable = false)
    private LocalDateTime createdAt;
}

File: frontend/src/features/suppliers/types/supplier.ts
export interface SupplierResponse {
    id: number;
    name: string;
    nip: string;
    email: string;
    phone: string;
    street: string;
    city: string;
    postalCode: string;
    country: string; 
    active: boolean;
    createdAt: string;
}

export interface SupplierRequest {
    name: string;
    nip: string;
    email: string;
    phone: string;
    street: string;
    city: string;
    postalCode: string;
    country?: string;
}

export interface Page<T> {
    content: T[];
    totalElements: number;
    totalPages: number;
    size: number;
    number: number;
}

export interface SupplierListParams {
    search?: string;
    active?: boolean;
    page: number;
    size: number;
    sort?: string;
}

